File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-shared-5-4/SettlementInformation.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class SettlementInformation : ISerialized
    {
        public SettlementInformation(XmlNode xmlNode)
        {
            XmlNode standardSettlementStyleNode = xmlNode.SelectSingleNode("standardSettlementStyle");
            
            if (standardSettlementStyleNode != null)
            {
                if (standardSettlementStyleNode.Attributes["href"] != null || standardSettlementStyleNode.Attributes["id"] != null) 
                {
                    if (standardSettlementStyleNode.Attributes["id"] != null) 
                    {
                        standardSettlementStyleIDRef_ = standardSettlementStyleNode.Attributes["id"].Value;
                        StandardSettlementStyleEnum ob = new StandardSettlementStyleEnum(standardSettlementStyleNode);
                        IDManager.SetID(standardSettlementStyleIDRef_, ob);
                    }
                    else if (standardSettlementStyleNode.Attributes["href"] != null)
                    {
                        standardSettlementStyleIDRef_ = standardSettlementStyleNode.Attributes["href"].Value;
                    }
                    else
                    {
                        standardSettlementStyle_ = new StandardSettlementStyleEnum(standardSettlementStyleNode);
                    }
                }
                else
                {
                    standardSettlementStyle_ = new StandardSettlementStyleEnum(standardSettlementStyleNode);
                }
            }
            
        
            XmlNode settlementInstructionNode = xmlNode.SelectSingleNode("settlementInstruction");
            
            if (settlementInstructionNode != null)
            {
                if (settlementInstructionNode.Attributes["href"] != null || settlementInstructionNode.Attributes["id"] != null) 
                {
                    if (settlementInstructionNode.Attributes["id"] != null) 
                    {
                        settlementInstructionIDRef_ = settlementInstructionNode.Attributes["id"].Value;
                        SettlementInstruction ob = new SettlementInstruction(settlementInstructionNode);
                        IDManager.SetID(settlementInstructionIDRef_, ob);
                    }
                    else if (settlementInstructionNode.Attributes["href"] != null)
                    {
                        settlementInstructionIDRef_ = settlementInstructionNode.Attributes["href"].Value;
                    }
                    else
                    {
                        settlementInstruction_ = new SettlementInstruction(settlementInstructionNode);
                    }
                }
                else
                {
                    settlementInstruction_ = new SettlementInstruction(settlementInstructionNode);
                }
            }
            
        
        }
        
    
        #region StandardSettlementStyle_
        private StandardSettlementStyleEnum standardSettlementStyle_;
        public StandardSettlementStyleEnum StandardSettlementStyle_
        {
            get
            {
                if (this.standardSettlementStyle_ != null)
                {
                    return this.standardSettlementStyle_; 
                }
                else if (this.standardSettlementStyleIDRef_ != null)
                {
                    standardSettlementStyle_ = IDManager.getID(standardSettlementStyleIDRef_) as StandardSettlementStyleEnum;
                    return this.standardSettlementStyle_; 
                }
                else
                {
                      return this.standardSettlementStyle_; 
                }
            }
            set
            {
                if (this.standardSettlementStyle_ != value)
                {
                    this.standardSettlementStyle_ = value;
                }
            }
        }
        #endregion
        
        public string standardSettlementStyleIDRef_ { get; set; }
        #region SettlementInstruction_
        private SettlementInstruction settlementInstruction_;
        public SettlementInstruction SettlementInstruction_
        {
            get
            {
                if (this.settlementInstruction_ != null)
                {
                    return this.settlementInstruction_; 
                }
                else if (this.settlementInstructionIDRef_ != null)
                {
                    settlementInstruction_ = IDManager.getID(settlementInstructionIDRef_) as SettlementInstruction;
                    return this.settlementInstruction_; 
                }
                else
                {
                      return this.settlementInstruction_; 
                }
            }
            set
            {
                if (this.settlementInstruction_ != value)
                {
                    this.settlementInstruction_ = value;
                }
            }
        }
        #endregion
        
        public string settlementInstructionIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-shared-5-4/SettlementInformation.cpp
// SettlementInformation.cpp 
#include "SettlementInformation.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

SettlementInformation::SettlementInformation(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //standardSettlementStyleNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* standardSettlementStyleNode = xmlNode->FirstChildElement("standardSettlementStyle");

   if(standardSettlementStyleNode){standardSettlementStyleIsNull_ = false;}
   else{standardSettlementStyleIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- standardSettlementStyleNode , address : " << standardSettlementStyleNode << std::endl;
   #endif
   if(standardSettlementStyleNode)
   {
      if(standardSettlementStyleNode->Attribute("href") || standardSettlementStyleNode->Attribute("id"))
      {
          if(standardSettlementStyleNode->Attribute("id"))
          {
             standardSettlementStyleIDRef_ = standardSettlementStyleNode->Attribute("id");
             standardSettlementStyle_ = boost::shared_ptr<StandardSettlementStyleEnum>(new StandardSettlementStyleEnum(standardSettlementStyleNode));
             standardSettlementStyle_->setID(standardSettlementStyleIDRef_);
             IDManager::instance().setID(standardSettlementStyleIDRef_,standardSettlementStyle_);
          }
          else if(standardSettlementStyleNode->Attribute("href")) { standardSettlementStyleIDRef_ = standardSettlementStyleNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { standardSettlementStyle_ = boost::shared_ptr<StandardSettlementStyleEnum>(new StandardSettlementStyleEnum(standardSettlementStyleNode));}
   }

   //settlementInstructionNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* settlementInstructionNode = xmlNode->FirstChildElement("settlementInstruction");

   if(settlementInstructionNode){settlementInstructionIsNull_ = false;}
   else{settlementInstructionIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- settlementInstructionNode , address : " << settlementInstructionNode << std::endl;
   #endif
   if(settlementInstructionNode)
   {
      if(settlementInstructionNode->Attribute("href") || settlementInstructionNode->Attribute("id"))
      {
          if(settlementInstructionNode->Attribute("id"))
          {
             settlementInstructionIDRef_ = settlementInstructionNode->Attribute("id");
             settlementInstruction_ = boost::shared_ptr<SettlementInstruction>(new SettlementInstruction(settlementInstructionNode));
             settlementInstruction_->setID(settlementInstructionIDRef_);
             IDManager::instance().setID(settlementInstructionIDRef_,settlementInstruction_);
          }
          else if(settlementInstructionNode->Attribute("href")) { settlementInstructionIDRef_ = settlementInstructionNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { settlementInstruction_ = boost::shared_ptr<SettlementInstruction>(new SettlementInstruction(settlementInstructionNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<StandardSettlementStyleEnum> SettlementInformation::getStandardSettlementStyle()
{
   if(!this->standardSettlementStyleIsNull_){
        if(standardSettlementStyleIDRef_ != ""){
             return boost::shared_static_cast<StandardSettlementStyleEnum>(IDManager::instance().getID(standardSettlementStyleIDRef_));
        }else{
             return this->standardSettlementStyle_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<StandardSettlementStyleEnum>();
   }
}
boost::shared_ptr<SettlementInstruction> SettlementInformation::getSettlementInstruction()
{
   if(!this->settlementInstructionIsNull_){
        if(settlementInstructionIDRef_ != ""){
             return boost::shared_static_cast<SettlementInstruction>(IDManager::instance().getID(settlementInstructionIDRef_));
        }else{
             return this->settlementInstruction_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<SettlementInstruction>();
   }
}
}

